DDF-1467: Update UrlResourceReader to securely handle file protocol

figliold opened the pull request:
@pklinef @garrettfreibott @lessarderic @andrew-fiedler

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf/149)
<!-- Reviewable:end -->

stustison commented on catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java:
```diff
@@ -77,23 +80,16 @@
-        qualifierSet.add(URL_FILE_SCHEME);
-    }
+    
+    private final Set<String> qualifierSet = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(new String[] {URL_HTTP_SCHEME, URL_HTTPS_SCHEME, URL_FILE_SCHEME})));
```
You don't need to make a string array here. Aslist takes individual object args as well.

stustison commented on catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java:
```diff
@@ -154,6 +151,31 @@ public MimeTypeMapper getMimeTypeMapper() {
+    public void setRootResourceDirectories(Set<String> rootResourceDirectoryPaths) {
+        if (rootResourceDirectoryPaths == null || rootResourceDirectoryPaths.isEmpty()) {
+            LOGGER.debug("rootResourceDirectoryPaths is null or empty. Clearing rootResourceDirectories.");
+            this.rootResourceDirectories.clear();
```
If rootResourceDirectories is null, then this is a null pointer exception at this point and if it is empty then clearing it does nothing.

stustison replied:
Ah nm. Read it wrong :)

andrew-fiedler replied:
If you have to clear out the ```rootResourceDirectories``` no matter what, you could just move it out of the ```if ```statement.  Then instead of an ```if else```, you could simply do an ```if``` surrounding the ```for``` loop.

stustison commented on catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java:
```diff
@@ -370,6 +398,40 @@ private void skipBytes(InputStream is, String bytesToSkip) throws IOException {
     }
+        
+    private boolean validateFilePath(File resourceFilePath) throws IOException {
+        String resourceCanonicalPath = resourceFilePath.getCanonicalPath();
```
Does canonical path remove stuff like /../../.. From the path so that something like /blah/../../../etc/passwd wouldn't match /blah and allow someone to download some file they shouldn't get to?

garrettfreibott commented:
<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 403 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyA7ugMrfsD6zJFi2lW-r1-403)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L403)):</sup>
@scott I believe that File's getCanonicalPath method evaluates .'s and ..'s, so something like /blah/blahblah/../../etc/passwd would be returned as /etc/passwd, which then should properly fail the root directory match below. There is also a new test in TestFederation with this PR that checks for this situation

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/149)
<!-- Sent from Reviewable.io -->

lessarderic commented:
Would have been nice to put the path validation logic in a separate class. Would have made that code potentially re-usable and would have kept the URLResourceReader code and unit tests simpler.

---

Review status: 0 of 15 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 85 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyA7ugMrfsD6zJFi2lW-r1-85)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L85)):</sup>
Recommend using Guava's ImmutableSet.of() instead if we can (see http://docs.guava-libraries.googlecode.com/git-history/release/javadoc/com/google/common/collect/ImmutableSet.html)

Also, could this be made static?

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 143 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyDytWo_OePWsPaZjoX)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L143)):</sup>
Since this never changes, should we log this every time the getter is called or only once when the class is instantiated?

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 158 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyA7ugMrfsD6zJFi2lW-r1-158)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L158)):</sup>
Since this setter does more than just set a value, can we add some JavaDoc and explain what happens when rootResourceDirectoryPath is null or empty?

Should also mention what format we expect those paths to be in and what happens if one of them is invalid.

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 164 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyE-gOceKor1jgdFuYt)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L164)):</sup>
What should we do if a Paths.get() throws an exception? Fail fast or best effort? Either way, we should at least log something and have unit tests for it.

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 418 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyE4VmRKQVlLkkRTqfu)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L418)):</sup>
Why not just return true and drop the isValid variable?

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/test/java/ddf/catalog/resource/impl/ResourceReaderTest.java, line 153 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyE0AKSNku5t0SznIHC)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/test/java/ddf/catalog/resource/impl/ResourceReaderTest.java#L153)):</sup>
Use Guava's ImmutableSet.of() if possible.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/149)
<!-- Sent from Reviewable.io -->

stustison commented:
Review status: 0 of 15 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 403 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyA7ugMrfsD6zJFi2lW-r1-403)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L403)):</sup>
:+1: 

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/149)
<!-- Sent from Reviewable.io -->

lessarderic commented:
Review status: 0 of 15 files reviewed at latest revision, 21 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 437 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEEXDG52SP3OVifTi8)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L437)):</sup>
Can we make this package-private instead?

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java, line 68 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEBfUl-JazrpaCLAig)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java#L68)):</sup>
ImmutableSet.of() and constant?

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java, line 77 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEIFffvhhhitxVhzCq)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java#L77)):</sup>
Any reason we have a default ctor _and_ setters for the ResourceReader and Tika? Should creating a `OgcUrlResourceReader` with null ResourceReader and/or Tika be allowed?

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java, line 91 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEBnZzS55no7hTDXIa)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java#L91)):</sup>
Log in ctor?

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java, line 102 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEBskgzxJIqJpzxqjg)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java#L102)):</sup>
Description?

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java, line 98 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEGKVZ2lNrerE0WyJD)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java#L98)):</sup>
Can we be more specific than `any(URL.class)` (same question for the following tests)?

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java, line 126 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEGTag73bGGb7GPRT_)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java#L126)):</sup>
There's a bunch of empty lines with spaces in them in most of the files. You may want to run the formatter before your next commit.

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java, line 321 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEHQX02hgLqJZAJYt1)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java#L321)):</sup>
This setup code is repeated in a bunch of methods. Any way we could extract a private method?

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java, line 360 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEJ11no6Bwo1BDrkIo)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java#L360)):</sup>
If we keep the default ctor, I'd use that instead of passing in nulls in these tests.

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java, line 383 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEJInk0r6RlbbbgysP)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java#L383)):</sup>
If the name _always has_ to be the class name, I'd do `assertThat(resourceReader.getId(), equalTo(OgcUrlResourceReader.class.getSimpleName()));`. Otherwise, `assertThat(resourceReader.getId(), equalTo(OgcUrlResourceReader.SHORTNAME));` is  probably good enough.

Same comment applies to the next 3 test methods.

---

<sup>**[distribution/ddf-common/pom.xml, line 65 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEKJzqSA5Bfrf_c5Pg)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/distribution/ddf-common/pom.xml#L65)):</sup>
Tab.

---

<sup>**[distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java, line 670 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEKgJCL1_WHKNLIrdk)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java#L670)):</sup>
Should be able to replace the loop with `ImmutableSet.of(rootResourceDirs);`.

---

<sup>**[distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java, line 245 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyELYFoZpRICsVWN_zB)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java#L245)):</sup>
Can't we return something else than a 200 status code?

---

<sup>**[distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java, line 248 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEMwH6EUCKrSJzypoE)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java#L248)):</sup>
Should we do this in the teardown or setup method to make sure it always gets done even if the test fails?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/149)
<!-- Sent from Reviewable.io -->

garrettfreibott commented:
Reviewed 15 of 15 files at r1.
Review status: all files reviewed at latest revision, 21 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/149)
<!-- Sent from Reviewable.io -->

pklinef commented on catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java:
```diff
@@ -156,19 +158,63 @@ public void setMimeTypeMapper(MimeTypeMapper mimeTypeMapper) {
+                    LOGGER.error("{} is an invalid path.", rootResourceDirectoryPath, e);
+                }
+            }
+        }
```
If rootResourceDirectoryPaths is null or blank, we could use KARAF_HOME as the default value.

pklinef commented on distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java:
```diff
@@ -217,13 +330,20 @@ public void testFederatedRetrieveExistingProductCsw() throws Exception {
     public void testFederatedRetrieveNoProductCsw() throws Exception {
+        File[] rootDirectories = File.listRoots();        
+        String rootDir = rootDirectories[0].getCanonicalPath();
+        urlResourceReaderConfigurator.setUrlResourceReaderRootDirs(new String[] {rootDir});
```
If you do not make KARAF_HOME the default, you could add a method like useKarafHome to simplify configuring this.

figliold commented:
I thought about that but it seemed like overkill since the logic is only in one method.  I'm going to leave it the way it is.

---

Review status: 7 of 18 files reviewed at latest revision, 23 unresolved discussions, some commit checks broke.

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 87 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyA7ugMrfsD6zJFi2lW-r1-85)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L87)):</sup>
fixed

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 149 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyDytWo_OePWsPaZjoX)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L149)):</sup>
fixed

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 171 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyA7ugMrfsD6zJFi2lW-r1-158)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L171)):</sup>
fixed

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 178 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyE-gOceKor1jgdFuYt)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L178)):</sup>
fixed

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 189 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyA7ugMrfsD6zJFi2lW-r2-189)** ([raw file](https://github.com/codice/ddf/blob/9ba909189a82e20f6e795e17bac407fd0973dd67/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L189)):</sup>
the default is KARAF_HOME/data/products

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 424 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyA7ugMrfsD6zJFi2lW-r1-403)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L424)):</sup>
yes it does

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 443 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyE4VmRKQVlLkkRTqfu)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L443)):</sup>
fixed

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java, line 461 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEEXDG52SP3OVifTi8)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L461)):</sup>
No, it is also used in the OgcUrlResourceReader

---

<sup>**[catalog/core/catalog-core-urlresourcereader/src/test/java/ddf/catalog/resource/impl/ResourceReaderTest.java, line 154 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyE0AKSNku5t0SznIHC)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/core/catalog-core-urlresourcereader/src/test/java/ddf/catalog/resource/impl/ResourceReaderTest.java#L154)):</sup>
fixed

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java, line 69 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEBfUl-JazrpaCLAig)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java#L69)):</sup>
fixed

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java, line 76 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEIFffvhhhitxVhzCq)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java#L76)):</sup>
fixed

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java, line 84 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEBnZzS55no7hTDXIa)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java#L84)):</sup>
fixed

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java, line 102 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEBskgzxJIqJpzxqjg)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/main/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/OgcUrlResourceReader.java#L102)):</sup>
fixed

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java, line 91 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEGKVZ2lNrerE0WyJD)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java#L91)):</sup>
fixed

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java, line 118 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEGTag73bGGb7GPRT_)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java#L118)):</sup>
fixed

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java, line 276 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEHQX02hgLqJZAJYt1)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java#L276)):</sup>
fixed

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java, line 329 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEJ11no6Bwo1BDrkIo)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java#L329)):</sup>
removed default ctor

---

<sup>**[catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java, line 352 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEJInk0r6RlbbbgysP)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/catalog/spatial/ogc/spatial-ogc-urlresourcereader/src/test/java/org/codice/ddf/spatial/ogc/catalog/resource/impl/TestResourceReader.java#L352)):</sup>
fixed unit tests

---

<sup>**[distribution/ddf-common/pom.xml, line 65 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEKJzqSA5Bfrf_c5Pg)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/distribution/ddf-common/pom.xml#L65)):</sup>
removed

---

<sup>**[distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java, line 475 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEKgJCL1_WHKNLIrdk)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java#L475)):</sup>
fixed

---

<sup>**[distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java, line 269 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyELYFoZpRICsVWN_zB)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java#L269)):</sup>
fixed

---

<sup>**[distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java, line 270 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyEMwH6EUCKrSJzypoE)** ([raw file](https://github.com/codice/ddf/blob/81d44786f7acf94de2e634f959bedf129ffc80cf/distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java#L270)):</sup>
fixed

---

<sup>**[distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java, line 346 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/149#-JyA7ugNKfixOeewqFSL-r2-346)** ([raw file](https://github.com/codice/ddf/blob/9ba909189a82e20f6e795e17bac407fd0973dd67/distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestFederation.java#L346)):</sup>
it defaults to KARAF_HOME/data/products

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/149)
<!-- Sent from Reviewable.io -->

lessarderic commented:
Reviewed 3 of 11 files at r2.
Review status: 10 of 18 files reviewed at latest revision, 5 unresolved discussions, some commit checks broke.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/149)
<!-- Sent from Reviewable.io -->
